### PR TITLE
Challenge 4 - zombie outbreak simulation

### PR DESCRIPTION
**Pull request by harrymunro**
# Zombie Outbreak Simulation

## Challenge Overview

The objective is to simulate the spread of a zombie outbreak in a grid-based city.
Each cell in the grid can either be empty, contain a human, or contain a zombie.
Each day, zombies spread to adjacent cells (up, down, left, right), turning any humans in those cells into zombies.
The challenge will be to write a function that performs the simulation for a given number of days and generates expected results.

## Instructions for Participants

    Understand the Grid Representation: The grid is a list of lists, where each element can be:
        'H' for a human
        'Z' for a zombie
        'E' for an empty cell (if needed)

    Simulation Logic:
        Each day, every zombie spreads to its adjacent cells (up, down, left, right).
        If an adjacent cell contains a human, it turns into a zombie.
        The simulation runs for a given number of days.

    Task:
        Implement the simulate_zombie_outbreak function.
        Ensure that the function correctly simulates the outbreak for the specified number of days.

## Bonus challenges:
- Can you add more complexity and/or scale to the simulation? e.g. randomness, humans fighting back or a larger grid?.
- If you add more complexity and find the behaviour becomes more "emergent", how would you adjust your testing approach?
- How could you animate your results? (hint: using Pygame or Matplotlib)

### CODE REVIEW

**Comment by harrymunro** [comment]
Thanks for the review @gtm19

File: 004/test_zombie_sim.py
Comment (gtm19) [inline 004/test_zombie_sim.py:13]: @harrymunro added some empties, as per the challenge description. Just adds a tiny bit more complexity to creating the solution, as participants will need to advance the simulation differently depending on the contents of the cell

File: 004/test_zombie_sim.py
Comment (gtm19) [inline 004/test_zombie_sim.py:45]: @harrymunro just added `pytest` here as you get a bit nicer output when running the file

File: README.md
Comment (gtm19) [inline README.md:50]: @harrymunro added a line to the readme

**Review by gtm19 — APPROVED** [review]
@harrymunro think this is a really good challenge - not too complex so very inclusive, but participants can make it harder for themselves if they wish.

**Comment by gtm19** [comment]
> Thanks for the review @gtm19

No problem @harrymunro 

Have assigned this to you so (I hope?!) you will be able to merge the pull request if you are happy. If not, let me know and I will merge.

**Comment by harrymunro** [comment]
@gtm19 changes look good to me. I can't merge the PR though as I don't have write access to the repo.
image

**Pull request merged**
